docs/src/userguide/source_files_and_compilation.rst

```diff
@@ -338,6 +338,15 @@ them through :func:`cythonize`::
 Distributing Cython modules
 ----------------------------
 
+.. note::
+
+    This section is historical and contains recommendations that outlived their
+    purpose. Following recent improvements in the distribution toolchain, it is
+    now recommended to *not* include generated files in distributions,
+    and require Cython at build-time as defined in
+    `PEP 518 <https://www.python.org/dev/peps/pep-0518/>`_
+    and `PEP 621 <https://www.python.org/dev/peps/pep-0621/>`_.
+
 It is strongly recommended that you distribute the generated ``.c`` files as well
```

Comment:
I think we can keep much of this section, just with a more optional wording.
```suggestion
It is possible to distribute the generated ``.c`` files as well
```
etc.